x-pack/plugins/security/server/authentication/api_key.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export interface CreateApiKeyOptions {
+  callCluster: any;
+  isSecurityFeatureDisabled: () => boolean;
+  body: {
+    name: string;
+    role_descriptors: Record<string, any>;
+    expiration?: string;
+  };
+}
+
+export interface CreateApiKeyResult {
+  id: string;
+  name: string;
+  expiration?: string;
+  api_key: string;
+}
+
+export async function createApiKey({
+  callCluster,
+  body,
+  isSecurityFeatureDisabled,
+}: CreateApiKeyOptions): Promise<CreateApiKeyResult | null> {
+  if (isSecurityFeatureDisabled()) {
+    return null;
+  }
+  return await callCluster('shield.createApiKey', { body });
```

Comment:
__nit:__ can you please leave a comment here explaining that user needs `manage_api_key` cluster privilege to use this API? Just for future reference.